portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java

```diff
@@ -1195,6 +1191,17 @@ public User addUserWithWorkflow(
 			workflowContext = Collections.emptyMap();
 		}
 
+		workflowServiceContext.setAttributes(
+			new HashMap<String, Serializable>());
+
+		workflowServiceContext.setAttribute("autoPassword", autoPassword);
+		workflowServiceContext.setAttribute("sendEmail", sendEmail);
+
+		if (autoPassword) {
+			workflowServiceContext.setAttribute(
+				"passwordUnencrypted", password1);
```

Comment:
Please don't store the password into DB, regardless of `autoPassword`.

Reply:
Hi @topolik, I have since came up with another solution, and sent the pull
https://github.com/pei-jung/liferay-portal/pull/639